cli/src/cli.rs

```diff
@@ -2297,6 +2374,72 @@ mod tests {
         path
     }
 
+    #[test]
+    fn test_generate_unique_signers() {
+        let CliSignerInfo {
+            signers,
+            indexes,
+            require_default_keypair,
+        } = generate_unique_signers(vec![]);
+        assert_eq!(signers.len(), 0);
+        assert_eq!(indexes.len(), 0);
+        assert_eq!(require_default_keypair, true);
+
+        let CliSignerInfo {
+            signers,
+            indexes,
+            require_default_keypair,
+        } = generate_unique_signers(vec![None, None]);
+        assert_eq!(signers.len(), 0);
+        assert_eq!(indexes, vec![0, 0]);
+        assert_eq!(require_default_keypair, true);
+
+        let keypair0 = keypair_from_seed(&[1u8; 32]).unwrap();
+        let keypair0_clone = keypair_from_seed(&[1u8; 32]).unwrap();
+        let signers = vec![None, Some(keypair0.into()), Some(keypair0_clone.into())];
+        let CliSignerInfo {
+            signers,
+            indexes,
+            require_default_keypair,
+        } = generate_unique_signers(signers);
+        assert_eq!(signers.len(), 1);
+        assert_eq!(indexes, vec![0, 1, 1]);
+        assert_eq!(require_default_keypair, true);
+
+        let keypair0 = keypair_from_seed(&[1u8; 32]).unwrap();
+        let keypair0_clone = keypair_from_seed(&[1u8; 32]).unwrap();
+        let signers = vec![Some(keypair0.into()), Some(keypair0_clone.into())];
+        let CliSignerInfo {
+            signers,
+            indexes,
+            require_default_keypair,
+        } = generate_unique_signers(signers);
+        assert_eq!(signers.len(), 1);
+        assert_eq!(indexes, vec![0, 0]);
+        assert_eq!(require_default_keypair, false);
+
+        // Signers with the same pubkey are not distinct
+        let keypair0 = Keypair::new();
```

Comment:
Could use a different seed with `keypair_from_seed()` here.  Then we get consistent output and lower cognitive load when debugging.